components/layout_2020/flow/inline.rs

```diff
@@ -2143,13 +2144,19 @@ impl IndependentFormattingContext {
     /// Picks either the first or the last baseline, depending on `baseline-source`.
     /// <https://drafts.csswg.org/css-inline/#baseline-source>
     fn pick_baseline(&self, baselines: &Baselines) -> Option<Au> {
-        // TODO: Currently this only supports the initial `baseline-source: auto`.
-        if let Self::NonReplaced(non_replaced) = self {
-            if let NonReplacedFormattingContextContents::Flow(_) = non_replaced.contents {
-                return baselines.last;
-            }
+        match self.style().clone_baseline_source() {
+            BaselineSource::First => baselines.first,
+            BaselineSource::Last => baselines.last,
+            BaselineSource::Auto => {
+                // The existing logic for `Auto` remains unchanged.
```

Comment:
You can also remove this comment.

Reply:
That's what I thought. Will remove.

Reply:
It's not removed.

Reply:
It's my bad. I removed it. But while resolving conflicts. I accidently added it.